arrows/vxl/vidl_ffmpeg_video_input.cxx

```diff
@@ -373,8 +380,42 @@ class vidl_ffmpeg_video_input::priv
     return retval;
   } // misp_time
 
+  // -----------------------------------------------------------------------------------------------
+  bool presentation_time()
+  {
+    int frame_count( c_time_scan_frame_limit );
+
+    meta_ts = 0;
+    d_have_abs_frame_time = true;
+
+    while ( ( meta_ts == 0.0 )
+            && d_video_stream.advance()
+            && (( c_time_scan_frame_limit == 0) || frame_count-- ));
+
+    return true;
+  } // presentation_time
+
+  // -----------------------------------------------------------------------------------------------
+  bool current_time()
+  {
+    int frame_count( c_time_scan_frame_limit );
+
+    using namespace std::chrono;
+    auto current_time = system_clock::now();
+    auto current_time_ms = time_point_cast<microseconds>(current_time);
+
+    meta_ts = current_time_ms.time_since_epoch().count();
+
+    d_have_abs_frame_time = true;
 
-// ------------------------------------------------------------------
+    while ( ( meta_ts == 0.0 )
```

Comment:
This loop is not doing anything except slowing down your program when you open a video.

Reply:
I'll remove it